pandas/core/missing.py

```diff
@@ -623,7 +665,10 @@ def interpolate_2d(
             raise AssertionError("cannot interpolate on a ndim == 1 with axis != 0")
         values = values.reshape(tuple((1,) + values.shape))
 
-    method = clean_fill_method(method)
+    method_cleaned = clean_fill_method(method)
+    assert isinstance(method_cleaned, str)
```

Comment:
Another alternative could be to remove the `None` return from `clean_fill_method`, and thus type that function to always return `str`. 
That might need a `if method is not None: ` in `generic.py` align method, but in most places where this function is called, we already know the input is a string (and thus not None)